src/game/AI/ScriptDevAI/scripts/outland/hellfire_citadel/shattered_halls/shattered_hallsScripts.cpp

```diff
@@ -270,6 +269,251 @@ struct npc_shattered_hand_centurion : public CombatAI
+                closest->GetMotionMaster()->MovePoint(100, FelOrcSpawnCoords[legionnaireGuid][0], FelOrcSpawnCoords[legionnaireGuid][1], FelOrcSpawnCoords[legionnaireGuid][2], FORCED_MOVEMENT_RUN);
+            }
+        }
+        else if (m_creature->HasStringId(SEVENTH_LEGIONNAIRE_STRING))
+        {
+            // For the legionnaire 07, if one of his group members dies he will inform the nearest npc staying at the dummys behind him
+            // all 3 npc at dummys have StringID 5400015 assigned
+            auto m_dummyReinf = m_creature->GetMap()->GetCreatures(DUMMY_REINF_STRING_1);
+            Creature* closest = nullptr;
+            for (Creature* creature : *m_dummyReinf)
+            {
+                // Only call alive creatures
+                // Only call creature that isnt in combat already
+                if (creature->IsAlive() && !creature->IsInCombat())
+                    continue;
+
+                if (!closest)
+                    closest = creature;
+                else if (m_creature->GetDistance(creature, true, DIST_CALC_NONE) < m_creature->GetDistance(closest, true, DIST_CALC_NONE))
+                    closest = creature;
+            }
+            if (closest)
+            {
+                closest->GetMotionMaster()->MovePoint(100, m_creature->GetPositionX(), m_creature->GetPositionY(), m_creature->GetPositionZ(), FORCED_MOVEMENT_RUN);
+                DoBroadcastText(aRandomReinf[urand(0, 6)], m_creature);
+            }
+        }
+        else if (m_creature->HasStringId(EIGTH_LEGIONNAIRE_STRING))
+        {
+            // For the legionnaire 08, if one of his group members dies he will inform the nearest npc staying at the dummys behind him
+            // all 3 npc at dummys have StringID 5400016 assigned
+            auto m_dummyReinf = m_creature->GetMap()->GetCreatures(DUMMY_REINF_STRING_2);
+            WorldObject* closest = nullptr;
+            for (Creature* creature : *m_dummyReinf)
+            {
+                // Only call alive creatures
+                // Only call creature that isnt in combat already
+                if (creature->IsAlive() && !creature->IsInCombat())
+                    continue;
+
+                if (!closest)
+                    closest = creature;
+                else if (m_creature->GetDistance(creature, true, DIST_CALC_NONE) < m_creature->GetDistance(closest, true, DIST_CALC_NONE))
+                    closest = creature;
+            }
+            if (closest)
+            {
+                static_cast<Creature*>(closest)->GetMotionMaster()->MovePoint(100, m_creature->GetPositionX(), m_creature->GetPositionY(), m_creature->GetPositionZ(), FORCED_MOVEMENT_RUN);
+                DoBroadcastText(aRandomReinf[urand(0, 6)], m_creature);
+            }
+        }
+
+        // Buff can only get casted when legionnaire is infight and doesnt already have the buff
+        if (!m_creature->SelectHostileTarget() || !m_creature->GetVictim())
```

Comment:
This code "m_creature->SelectHostileTarget()" should never be put anywhere in modern cmangos AI. It can cause taunt/threat and other things to behave unpredictably. GetVictim is fine and you can use IsInCombat instead. Though shouldnt SPELL_ENRAGE be in ExecuteAction?

Reply:
spell_enrage only gets used, when one of the group members around the legionnaire dies, and the legionnaire is also infight